src/Middleware/OutputCaching/src/OutputCacheKeyProvider.cs

```diff
@@ -166,6 +167,29 @@ public string CreateStorageKey(OutputCacheContext context)
                 }
             }
 
+            // Vary by route values
+            var routeValuesCount = varyByRules?.RouteValues.Count ?? 0;
+            if (routeValuesCount > 0)
+            {
+                // Append a group separator for the query key segment of the cache key
+                builder.Append(KeyDelimiter)
+                    .Append('R');
+
+                for (var i = 0; i < routeValuesCount; i++)
+                {
+                    // The lookup key can't be null
+                    var routeValueName = varyByRules!.RouteValues[i] ?? string.Empty;
+
+                    // RouteValues returns null if the key doesn't exist
+                    var routeValueValue = context.HttpContext.Request.RouteValues[routeValueName];
+
+                    builder.Append(KeyDelimiter)
+                        .Append(routeValueName)
+                        .Append('=')
+                        .Append(Convert.ToString(routeValueValue, CultureInfo.InvariantCulture));
```

Comment:
A route value is an `object` as opposed to headers and queries. I think we need to force the culture here for normalize the key.